packages/backend/src/tokens/keys.ts

```diff
@@ -0,0 +1,207 @@
+      if (jwk) {
+        delete cache[jwk.kid];
+      } else {
+        cache = {};
+      }
+    }, jwksCacheTtlInMs);
+  }
+}
+
+const LocalJwkKid = 'local';
+
+/**
+ *
+ * Loads a local PEM key usually from process.env and transform it to JsonWebKey format.
+ * The result is also cached on the module level to avoid unnecessary computations in subsequent invocations.
+ *
+ * @param {string} key
+ * @returns {JsonWebKey} key
+ */
+export function loadClerkJWKFromLocal(localKey?: string): JsonWebKey {
+  if (!getFromCache(LocalJwkKid)) {
+    if (!localKey) {
+      throw new TokenVerificationError({
+        action: TokenVerificationErrorAction.SetClerkJWTKey,
+        message: 'Missing local JWK.',
+        reason: TokenVerificationErrorReason.LocalJWKMissing,
+      });
+    }
+
+    // Notice:
+    // Currently Next.js in development mode cannot parse PEM key format, but it can parse JWKs.
+    // This is a simple way to convert our PEM keys to JWKs until the bug is resolved.
+    const RSA_PREFIX = 'MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA';
+    const RSA_SUFFIX = 'IDAQAB';
+
+    // JWK https://datatracker.ietf.org/doc/html/rfc7517
+    setInCache(
+      {
+        kid: 'local',
+        kty: 'RSA',
+        n: localKey
+          .slice(RSA_PREFIX.length, RSA_SUFFIX.length * -1)
+          .replace(/\+/g, '-')
+          .replace(/\//g, '_'),
+        e: 'AQAB',
+      },
+      -1, // local key never expires in cache
+    );
+  }
+
+  return getFromCache(LocalJwkKid);
+}
+
+export type LoadClerkJWKFromRemoteOptions = {
+  kid: string;
+  jwksCacheTtlInMs?: number;
+  skipJwksCache?: boolean;
+} & (
+  | {
+      apiKey: string;
+      apiUrl?: string;
+      apiVersion?: string;
+      issuer?: never;
+    }
+  | {
+      apiKey?: never;
+      apiUrl?: never;
+      apiVersion?: never;
+      issuer: string;
+    }
+);
+
+/**
+ *
+ * Loads a key from JWKS retrieved from the well-known Frontend API endpoint of the issuer.
+ * The result is also cached on the module level to avoid network requests in subsequent invocations.
+ * The cache lasts 1 hour by default.
+ *
+ * @param {Object} options
+ * @param {string} options.issuer - The issuer origin of the JWT
+ * @param {string} options.kid - The id of the key that the JWT was signed with
+ * @param {string} options.alg - The algorithm of the JWT
+ * @param {number} options.jwksCacheTtlInMs - The TTL of the jwks cache (defaults to 1 hour)
+ * @returns {JsonWebKey} key
+ */
+export async function loadClerkJWKFromRemote({
+  apiKey,
+  apiUrl,
+  apiVersion,
+  issuer,
+  kid,
+  jwksCacheTtlInMs = 1000 * 60 * 60, // 1 hour,
+  skipJwksCache,
+}: LoadClerkJWKFromRemoteOptions): Promise<JsonWebKey> {
+  if (skipJwksCache || !getFromCache(kid)) {
+    let fetcher;
+
+    if (apiUrl) {
+      fetcher = () => fetchJWKSFromBAPI(apiUrl, apiKey, apiVersion);
+    } else if (issuer) {
+      fetcher = () => fetchJWKSFromFAPI(issuer);
+    } else {
+      throw new TokenVerificationError({
```

Comment:
🙃 we can replace this nested if by moving the error up and using a ternary for the fetcher directly inside the callWithRetry call